starlette/middleware/gzip.py

```diff
@@ -48,12 +49,19 @@ async def send_with_gzip(self, message: Message) -> None:
             # Don't send the initial message until we've determined how to
             # modify the outgoing headers correctly.
             self.initial_message = message
+            self.content_encoding_set = any(
+                header == b"content-encoding"
+                for header, _ in message.get("headers", [])
+            )
         elif message_type == "http.response.body" and not self.started:
```

Comment:
Can we only change it here instead?
```suggestion
        elif message_type == "http.response.body" and self.content_encoding_set:
            if not self.started:
                await send(self.initial_message)
            await send(message)
        elif message_type == "http.response.body" and not self.started:
```